coreneuron/io/phase2.cpp

```diff
@@ -0,0 +1,1211 @@
+#include "coreneuron/io/phase2.hpp"
+#include "coreneuron/coreneuron.hpp"
+#include "coreneuron/sim/multicore.hpp"
+#include "coreneuron/io/nrn_checkpoint.hpp"
+#include "coreneuron/utils/nrnoc_aux.hpp"
+#include "coreneuron/network/partrans.hpp"
+#include "coreneuron/permute/cellorder.hpp"
+#include "coreneuron/permute/node_permute.h"
+#include "coreneuron/utils/vrecitem.h"
+#include "coreneuron/io/nrn_soa.hpp"
+
+int (*nrn2core_get_dat2_1_)(int tid,
+                            int& ngid,
+                            int& n_real_gid,
+                            int& nnode,
+                            int& ndiam,
+                            int& nmech,
+                            int*& tml_index,
+                            int*& ml_nodecount,
+                            int& nidata,
+                            int& nvdata,
+                            int& nweight);
+
+int (*nrn2core_get_dat2_2_)(int tid,
+                            int*& v_parent_index,
+                            double*& a,
+                            double*& b,
+                            double*& area,
+                            double*& v,
+                            double*& diamvec);
+
+int (*nrn2core_get_dat2_mech_)(int tid,
+                               size_t i,
+                               int dsz_inst,
+                               int*& nodeindices,
+                               double*& data,
+                               int*& pdata);
+
+int (*nrn2core_get_dat2_3_)(int tid,
+                            int nweight,
+                            int*& output_vindex,
+                            double*& output_threshold,
+                            int*& netcon_pnttype,
+                            int*& netcon_pntindex,
+                            double*& weights,
+                            double*& delays);
+
+int (*nrn2core_get_dat2_corepointer_)(int tid, int& n);
+
+int (*nrn2core_get_dat2_corepointer_mech_)(int tid,
+                                           int type,
+                                           int& icnt,
+                                           int& dcnt,
+                                           int*& iarray,
+                                           double*& darray);
+
+int (*nrn2core_get_dat2_vecplay_)(int tid, int& n);
+
+int (*nrn2core_get_dat2_vecplay_inst_)(int tid,
+                                       int i,
+                                       int& vptype,
+                                       int& mtype,
+                                       int& ix,
+                                       int& sz,
+                                       double*& yvec,
+                                       double*& tvec);
+
+namespace coreneuron {
+template <typename T>
+inline void mech_layout(T* data, int cnt, int sz, int layout) {
```

Comment:
```suggestion
inline void mech_data_transpose(T* data, int cnt, int sz, int layout) {
```
@pramodk can this function be renamed and wouldn't this be a better name?

Reply:
or `mech_matrix_transpose` ?

Reply:
I don't get what it does, so choose the new name yourself, I cannot have a good idea on that

Reply:
Done